ScoutSuite/providers/aws/facade/iam.py

```diff
@@ -0,0 +1,203 @@
+                    id_field = entity_type.replace('Policy', '')[:-1] + 'Id'
+                    resource_id = entity[id_field]
+                    policy['attached_to'][resource_type].append(
+                        {'name': resource_name, 'id': resource_id})
+
+        return policies
+
+    async def get_users(self):
+        client = AWSFacadeUtils.get_client('iam', self.session)
+        users = await AWSFacadeUtils.get_all_pages('iam', None, self.session, 'list_users', 'Users')
+
+        # TODO: Parallelize this
+        for user in users:
+            user_name = user['UserName']
+            user_id = user['UserId']
+
+            policies = self._get_inline_policies('user', user_id, user_name)
+            if len(policies):
+                user['inline_policies'] = policies
+            user['inline_policies_count'] = len(policies)
+            user['groups'] = []
+            groups = await AWSFacadeUtils.get_all_pages('iam', None, self.session, 'list_groups_for_user', 'Groups', UserName=user_name)
+            for group in groups:
+                user['groups'].append(group['GroupName'])
+            try:
+                user['LoginProfile'] = client.get_login_profile(UserName=user_name)[
+                    'LoginProfile']
+            except Exception:
+                pass
+            user['AccessKeys'] = await self._get_user_acces_keys(user_name)
+            user['MFADevices'] = await self._get_user_mfa_devices(user_name)
+
+        return users
+
+    async def get_roles(self):
+        roles = await AWSFacadeUtils.get_all_pages('iam', None, self.session, 'list_roles', 'Roles')
+
+        # Handle throttling errors
+        for role in roles:
+            role['instances_count'] = 'N/A'
+
+            # Get role policies
+            policies = self._get_inline_policies(
+                'role', role['RoleId'], role['RoleName'])
+            if len(policies):
+                role['inline_policies'] = policies
+            role['inline_policies_count'] = len(policies)
+
+            # Get instance profiles
+            profiles = await AWSFacadeUtils.get_all_pages('iam', None, self.session, 'list_instance_profiles_for_role', 'InstanceProfiles', RoleName=role['RoleName'])
+            role.setdefault('instance_profiles', {})
+            for profile in profiles:
+                profile_id = profile['InstanceProfileId']
+                role['instance_profiles'].setdefault(profile_id, {})
+                role['instance_profiles'][profile_id].setdefault(
+                    'arn', profile['Arn'])
+                role['instance_profiles'][profile_id].setdefault(
+                    'name', profile['InstanceProfileName'])
+
+            # Get trust relationship
+            role['assume_role_policy'] = {}
+            role['assume_role_policy']['PolicyDocument'] = role.pop(
+                'AssumeRolePolicyDocument')
+
+        return roles
+
+    async def get_password_policy(self):
```

Comment:
It was likely already like this before you moved it, but this method doesn't have a `_` in front of it while other getters have one it seems :sweat_smile:

Reply:
The `_` is a convention (defined in PEP 8) which means _"you shouldn't use this outside of this class"_, or _"for internal use only"_, by opposition. This getter is used by `PasswordPolicy`'s `fetch_all`, hence the non-prefixed name.